src/database/db_connection.rs

```diff
@@ -207,7 +217,9 @@ impl TransactionTrait for DatabaseConnection {
             DatabaseConnection::MockDatabaseConnection(conn) => {
                 DatabaseTransaction::new_mock(Arc::clone(conn), None).await
             }
-            DatabaseConnection::Disconnected => panic!("Disconnected"),
+            DatabaseConnection::Disconnected => {
+                Err(DbErr::Conn(RuntimeErr::Internal("Disconnected".to_owned())))
```

Comment:
Some thoughts. Maybe is better to store in `RuntimeErr::Internal` `Cow<'static, str>` instead `String`? And we don't need call: `to_owned`?

Reply:
Very cool! Great suggestion! I think this is actually an optimization over the use of `String`. `RuntimeErr::Internal(Cow<'static, str>)` requires no heap allocation given that it was constructed via a `&str`, i.e. `RuntimeErr::Internal("Disconnected")`.

Thoughts? @tyt2y3

Reply:
But still... we need `"xxx".into()`. Correct me if I get it wrong @ikrivosheev

Reply:
I don't think `Cow` is good to be part of the public API data structure. More like an internal implementation optimization.
And I think it's okay to allocate in Error paths.

Reply:
@tyt2y3 ok